client/src/App/ProtectedRoute.jsx

```diff
@@ -4,12 +4,11 @@ import PropTypes from 'prop-types';
 import { AuthContext } from '../context/AuthContext';
 
 function ProtectedRoute(props) {
-  const userInfo = useContext(AuthContext);
+  const { userData } = useContext(AuthContext);
   const { children } = props;
-  const { isAdmin } = userInfo;
 
   return (
-    isAdmin ? (
+    userData.role ? (
```

Comment:
What you want to say here is `userData.isAuthorized ? `, rather than using the role. Even if it made sense to your application, it should also be readable, because I'm sitting here looking at you checking if there is a userRole as means of authentication and not knowing anything about your logic behind it.

I'm not suggesting to rename the variable `role`, I'm suggesting you'd introduce a whole new boolean variable `isAuthorized` that handles these cases.